### PR TITLE
Feature/google workflows gha

### PR DESCRIPTION
**Pull request by huystuhh**
Setting up new GH Action to deploy to Google Workflows

### CODE REVIEW

File: .github/workflows/deploy_google_workflow.yaml [hunk 1]
```diff
@@ -14,24 +22,38 @@ jobs:
...
       with:
-        workload_identity_provider: 'projects/1031862664530/locations/global/workloadIdentityPools/github-actions/providers/github-actions'
-        service_account: 'github-actions@prod8b61f23e.iam.gserviceaccount.com'
-    - name: 'Set up Cloud SDK'
+        workload_identity_provider: $GOOGLE_WORKLOAD_IDENTITY_PROVIDER
+        service_account: $GOOGLE_SERVICE_ACCOUNT
+    # Get list of files in last commit
+    - id: 'files'
+      name: Get list of changed files
+      uses: jitterbit/get-changed-files@v1
```
Comment (michaelefisher) [inline .github/workflows/deploy_google_workflow.yaml:38]: Ohhhh that's cool.

**Review by michaelefisher — COMMENTED** [review]
So before we do this, just to confirm, this is going to deploy new versions of the workflows unless we adjust the names - right?

**Comment by huystuhh** [comment]
@michaelefisher it'll overwrite the existing ones if the filename == workflow ID, which they all should be since we don't have very many right now. So yes, they'll be new versions/revisions unless we change the name (in which case, we would have two workflows with the old + new name but that makes me think maybe we should delete with github actions if removed??)

File: .github/workflows/deploy_google_workflow.yaml [hunk 2]
```diff
@@ -14,24 +22,51 @@ jobs:
       contents: 'read'
       id-token: 'write'
     steps:
+    # Checkout repo, this needs to happen first
     - name: Checkout
       uses: actions/checkout@v3
+    # Keyless Auth!
     - id: 'auth'
-      name: 'Authenticate to Google Cloud'
+      name: Authenticate to Google Cloud
```
Comment (jitoquinto) [inline .github/workflows/deploy_google_workflow.yaml:30]: nice!

**Review by jitoquinto — APPROVED** [review]

File: .github/workflows/deploy_google_workflow.yaml [hunk 3]
```diff
@@ -14,24 +22,51 @@ jobs:
...
-    #     gcloud workflows deploy illumis-data-ingestion \
-    #     --source workflows/illumis/illumis-data-ingestion.workflows.yaml
\ No newline at end of file
+        mapfile -d ',' -t added_modified_files < <(printf '%s,' '${{ steps.files.outputs.added_modified }}')
+        for file in "${added_modified_files[@]}"; do
+          case ${file} in *.workflows.yaml)
+            file_name=$(basename ${file})
+            workflow_id=$(echo ${file_name%%.*})
+
+            gcloud workflows deploy ${workflow_id} \
```
Comment (michaelefisher) [inline .github/workflows/deploy_google_workflow.yaml:55]: @huystuhh we should also supply the following:

--project=$GOOGLE_PROJECT_ID \
--location=us-central1 \
--service-account=[SA]

I guess we can include a default SA for now?

Comment (michaelefisher) [reply .github/workflows/deploy_google_workflow.yaml:55]: Or am I misunderstanding this and deploying without an SA will not remove the existing SA the workflow is running as?

**Pull request merged**